src/test/java/GeneTest.java

```diff
@@ -0,0 +1,55 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GeneTest {
+
+  @Test
+  public void testArraySizeGeneWithMock() {
+    // Given
+
+    RandomProvider randomProviderMock = mock(RandomProvider.class);
+
+    // given
+    int numberExpected = 113;
+    when(randomProviderMock.getRandom(0, 2)).thenReturn(numberExpected);
+
+    // when
+    Gene actual = new Gene(randomProviderMock);
+
+    // then
+    assertEquals(numberExpected, actual.getValues().length);
+  }
+
+  @Test
+  public void testValueUnderTheIndexGeneWithMock() {
```

Comment:
I am missing the point of this test. We should check if every char in Gene.values is equal to expectedNumber. Why not to check it with for each loop? It should work for all elements in char.